source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -231,6 +233,49 @@ def _filter_max_successive_buses(journeys, request):
             mark_as_dead(j, "too_much_successive_buses")
 
 
+def _filter_too_much_changes(journeys, instance, request):
+    """
+    eliminates journeys with correspondences more then nb correspondence of best journey + _max_additional_changes
+    """
+    logger = logging.getLogger(__name__)
+    max_additional_changes = get_or_default(request, '_max_additional_changes',
+                                            instance.max_additional_changes)
+    max_correspondence_allowed = max_additional_changes + min_correspondence_count(journeys)
+    for j in journeys:
+        if _to_be_deleted(j):
+            continue
+
+        if get_correspondence_count(j) > max_correspondence_allowed:
+            logger.debug("the journey {} has a too much correspondences, we delete it".format(j.internal_id))
+            mark_as_dead(j, "too_much_correspondences")
+
+
+def min_correspondence_count(journeys):
+    """
+    Returns min correspondence count among journeys
+    """
+    nb_min_changes = 100
```

Comment:
I think you can rewrite this like:

```
if not journeys:
return None (and check for this case in the calling method)

return min(get_nb_connections(j) for j in journeys if not _to_be_deleted(j))
```
(might not 'compil' but it's the spirit)